### ISSUE TITLE
Ocasional crashs on iOS 9 and 10

### ISSUE THREAD

**Issue by labSacra** [Flipboard/FLAnimatedImage#159]
Hey guys,

I'm using this lib on a project of mine and I'm having ocasional crashs on iOS 9 and 10. 
The stack trace:

Crashed: com.flipboard.framecachingqueue
EXC_BAD_ACCESS KERN_INVALID_ADDRESS 0x000000010d07c000

0  libsystem_platform.dylib       0x18b51ee60 _platform_memmove + 96
1  ImageIO                        0x18e2d9a6c GIFReadPlugin::copyImageBlockSet(InfoRec*, CGImageProvider*, CGRect, CGSize, __CFDictionary const*) + 3244
2  ImageIO                        0x18e2d9a6c GIFReadPlugin::copyImageBlockSet(InfoRec*, CGImageProvider*, CGRect, CGSize, __CFDictionary const*) + 3244
3  ImageIO                        0x18e2d8788 GIFReadPlugin::CopyImageBlockSetProc(void*, CGImageProvider*, CGRect, CGSize, __CFDictionary const*) + 124
4  ImageIO                        0x18e13a8a4 IIOImageProviderInfo::copyImageBlockSetWithOptions(CGImageProvider*, CGRect, CGSize, __CFDictionary const*) + 496
5  ImageIO                        0x18e138830 IIOImageProviderInfo::CopyImageBlockSetWithOptions(void*, CGImageProvider*, CGRect, CGSize, __CFDictionary const*) + 356
6  CoreGraphics                   0x18da7b8f8 img_blocks_create + 260
7  CoreGraphics                   0x18da83328 img_data_lock + 1896
8  CoreGraphics                   0x18da82b6c CGSImageDataLock + 152
9  CoreGraphics                   0x18d89b8e8 ripc_AcquireImage + 712
10 CoreGraphics                   0x18da97404 ripc_DrawImage + 676
11 CoreGraphics                   0x18da874a8 CGContextDrawImageWithOptions + 636
12 APP                      0x1000c2f38 +[FLAnimatedImage predrawnImageFromImage:] (FLAnimatedImage.m:707)
13 APP                      0x1000c2474 -[FLAnimatedImage imageAtIndex:] (FLAnimatedImage.m:528)
14 APP                      0x1000c1fc8 __42-[FLAnimatedImage addFrameIndexesToCache:]_block_invoke_3 (FLAnimatedImage.m:452)
15 Foundation                     0x18cf30888 __NSIndexSetEnumerate + 824
16 APP                      0x1000c1f00 __42-[FLAnimatedImage addFrameIndexesToCache:]_block_invoke_2 (FLAnimatedImage.m:481)
17 libdispatch.dylib              0x18b31d200 _dispatch_call_block_and_release + 24
18 libdispatch.dylib              0x18b31d1c0 _dispatch_client_callout + 16
19 libdispatch.dylib              0x18b32b444 _dispatch_queue_serial_drain + 928
20 libdispatch.dylib              0x18b3209a8 _dispatch_queue_invoke + 652
21 libdispatch.dylib              0x18b32b940 _dispatch_queue_override_invoke + 360
22 libdispatch.dylib              0x18b32d38c _dispatch_root_queue_drain + 572
23 libdispatch.dylib              0x18b32d0ec _dispatch_worker_thread3 + 124
24 libsystem_pthread.dylib        0x18b5252c8 _pthread_wqthread + 1288
25 libsystem_pthread.dylib        0x18b524db4 start_wqthread + 4

I know EXC_BAD_ACCESS is normally memory leak. 
Any ideas on how to solve this problem?

Thanks in advance.

**Comment by yxztj** [comment]
The same here. Very rare(<0.1%) and hard to repro, but it's happening everyday in our production.

**Comment by geek-paulwong** [comment]
Use https://github.com/ibireme/YYWebImage it works great for me on GIF loading!!

**Comment by master-nevi** [comment]
Suggesting another library is not a solution to this problem. Bad form.

**Comment by geek-paulwong** [comment]
@master-nevi agree... except that this FLAnimatedImage is no longer actively maintained... and YYWebImage works perfectly, lol.

**Comment by flybrotherlixiang** [comment]
Same crash here.
